storage/s3.go

```diff
@@ -112,7 +112,6 @@ func (s *s3Storage) Put(ctx context.Context, path string, contentType string, co
 }
 
 func (s *s3Storage) batchWorker(ctx context.Context, uploads chan *Upload, errors chan error, stop chan bool, wg *sync.WaitGroup) {
-	wg.Add(1)
```

Comment:
if the goroutines don't start fast enough they don't add themselves to the workgroup before the main thread decides the work group is empty and carries on